File: Back_Tracking/C++/FibonacciSequence.cpp
#include <iostream>
using namespace std;

// Function to print Fibonacci sequence recursively
void findFibonacci(int n, int a, int b) {
    // Store current value of b
    int temp = b;
    // Calculate next Fibonacci number
    b = a + b;
    // Update a to previous b value
    a = temp;
    
    // Base case: stop if we exceed the limit
    if (n < a || n < b) {
        return;
    }
    
    // Print current Fibonacci number
    cout << "+" << b;
    // Recursive call for next number
    findFibonacci(n, a, b);
}

int main() {
    int n;
    // Prompt user for input
    cout << "Enter the Limit of fibonacci sequence: ";
    cin >> n;
    
    // Print initial sequence
    cout << "1+1";
    // Start Fibonacci sequence
    findFibonacci(n, 1, 1);
    cout << endl;
    
    return 0;
} 

File: Back_Tracking/Kotlin/FibonacciSequence.kt
import java.util.Scanner

// Function to print Fibonacci sequence recursively
fun findFibonacci(n: Int, a: Int, b: Int) {
    // Store current value of b
    var temp = b
    // Calculate next Fibonacci number
    var newB = a + b
    // Update a to previous b value
    var newA = temp
    
    // Base case: stop if we exceed the limit
    if (n < newA || n < newB) {
        return
    }
    
    // Print current Fibonacci number
    print("+$newB")
    // Recursive call for next number
    findFibonacci(n, newA, newB)
}

fun main() {
    // Create Scanner object for input
    val scanner = Scanner(System.`in`)
    
    // Prompt user for input
    print("Enter the Limit of fibonacci sequence: ")
    val n = scanner.nextInt()
    
    // Print initial sequence
    print("1+1")
    // Start Fibonacci sequence
    findFibonacci(n, 1, 1)
    println()
    
    // Close the scanner
    scanner.close()
} 